### PR TITLE
modify task cancel when time expired

### PR DESCRIPTION
**Pull request by llypb**
- [x]  logger changed
- [x]  Bancor parameter
- [x] Cancel transaction in time

### CODE REVIEW

File: src/AElf.Kernel/Miner/Application/MiningService.cs [hunk 1]
```diff
@@ -16,7 +17,17 @@ namespace AElf.Kernel.Miner.Application
 {
     public class MiningService : IMiningService
     {
-        public ILogger<MiningService> Logger { get; set; }
```
Comment (rosona) [inline src/AElf.Kernel/Miner/Application/MiningService.cs]: No need to change

File: test/AElf.Contracts.TestContract.Tests/AElf.Contracts.TestContract.Tests.csproj [hunk 2]
```diff
@@ -14,6 +14,7 @@
         <PackageReference Include="coverlet.msbuild" Version="2.5.1" />
         <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.2.0" />
         <PackageReference Include="Shouldly" Version="3.0.2" />
+        <PackageReference Include="System.Reactive" Version="4.1.6" />
```
Comment (rosona) [inline test/AElf.Contracts.TestContract.Tests/AElf.Contracts.TestContract.Tests.csproj]: Remove unused reference

File: src/AElf.Kernel.SmartContractExecution/Application/BlockExecutingService.cs [hunk 3]
```diff
@@ -44,19 +45,22 @@ public class BlockExecutingService : IBlockExecutingService, ITransientDependenc
             Logger.LogTrace("Entered ExecuteBlockAsync");
             var nonCancellable = nonCancellableTransactions.ToList();
             var cancellable = cancellableTransactions.ToList();
-
+            var stopWatch = new Stopwatch();
```
Comment (llypb) [inline src/AElf.Kernel.SmartContractExecution/Application/BlockExecutingService.cs]: to remove

File: src/AElf.Kernel/Miner/Application/MinerService.cs [hunk 4]
```diff
@@ -38,6 +38,10 @@ public class MinerService : IMinerService
             Duration blockExecutionTime)
         {
             var limit = await _blockTransactionLimitProvider.GetLimitAsync();
+//            var executableTransactionSet =
```
Comment (llypb) [inline src/AElf.Kernel/Miner/Application/MinerService.cs]: to remove

File: src/AElf.Kernel.SmartContract/Application/LocalTransactionExecutingService.cs [hunk 5]
```diff
@@ -116,6 +129,7 @@ private static bool IsTransactionCanceled(TransactionTrace trace)
             Transaction transaction, Timestamp currentBlockTime, CancellationToken cancellationToken,
             Address origin = null, bool isCancellable = true)
         {
+            await Task.Yield();
```
Comment (llypb) [inline src/AElf.Kernel.SmartContract/Application/LocalTransactionExecutingService.cs]: to add comment

File: src/AElf.Kernel.SmartContract/Application/LocalTransactionExecutingService.cs [hunk 6]
```diff
@@ -11,29 +11,35 @@
 using Google.Protobuf.WellKnownTypes;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Logging.Abstractions;
+using Volo.Abp.DependencyInjection;
 using Volo.Abp.EventBus.Local;
 
 namespace AElf.Kernel.SmartContract.Application
 {
-    public class TransactionExecutingService : ITransactionExecutingService
+    public interface ILocalTransactionExecutingService : ITransactionExecutingService
```
Comment (llypb) [inline src/AElf.Kernel.SmartContract/Application/LocalTransactionExecutingService.cs]: to remove interface

**Pull request merged**